lib/identity_cache.rb

```diff
@@ -89,8 +91,8 @@ def should_use_cache? # :nodoc:
     # == Parameters
     # +key+ A cache key string
     #
-    def fetch(key)
-      if should_use_cache?
+    def fetch(key, use_cache: should_use_cache?)
```

Comment:
I'm not sure I like this particular implementation, I think giving the client of the model the responsibility/power to bypass the cache a call at the time will end up in a lot of cognitive overhead on both implementers and reviewers. 

From reading the original issue it seems that you have context dependent needs to disable the cache, as in "I really want to make sure  that under this context model `M` does not try to use the cache".

What about a block syntax?

```
M.disable_cacheing do
end
```

or

```
IdentityCache.disable_cacheing_for(*M) do
end
```

Granted that would cause some surprise on the methods  called inside the  block, but would make it easier to separate contexts when you are reasoning about what should be cached and what should not.